src/cascadia/TerminalSettingsModel/ActionMap.cpp

```diff
@@ -15,36 +15,37 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
 {
     static InternalActionID Hash(const Model::ActionAndArgs& actionAndArgs)
     {
-        size_t hashedAction{ HashUtils::HashProperty(actionAndArgs.Action()) };
+        const auto action = actionAndArgs.Action();
+        til::hasher hasher;
 
-        size_t hashedArgs{};
-        if (const auto& args{ actionAndArgs.Args() })
+        if (const auto args = actionAndArgs.Args())
         {
-            // Args are defined, so hash them
-            hashedArgs = gsl::narrow_cast<size_t>(args.Hash());
+            hasher = gsl::narrow_cast<size_t>(args.Hash(hasher.finalize()));

```

Comment:
Before this is merged I'd like to make the `til::hasher constructor` explicit instead to prevent bugs.